client/state/reader/follows/reducer.js

```diff
@@ -67,6 +69,44 @@ function updatePostSubscription( state, { payload, type } ) {
 			...follow,
 			delivery_methods: {
 				email: newFollowState,
+				notification: get( follow, [ 'delivery_methods', 'notification' ], {} ),
+			},
+		},
+	};
+}
+
+function updateNotificationSubscription( state, { payload, type } ) {
+	const follow = find( state, { blog_ID: +payload.blogId } );
+	if ( ! follow ) {
+		return state;
+	}
+
+	const currentFollowState = get( follow, [ 'delivery_methods', 'notification' ], {} );
+
+	const newProps = {};
+	switch ( type ) {
+		case READER_SUBSCRIBE_TO_NEW_POST_NOTIFICATIONS:
+		case READER_UNSUBSCRIBE_TO_NEW_POST_NOTIFICATIONS:
+			newProps.send_posts = ! ( type === READER_UNSUBSCRIBE_TO_NEW_POST_NOTIFICATIONS );
+			break;
+	}
+
+	const newFollowState = {
```

Comment:
can newFollowState have other keys than `send_posts` within it?

Reply:
It can't, so I've simplified the function to just compare the `send_posts` boolean instead.